uspace/lib/c/generic/adt/hash_table.c

```diff
@@ -272,14 +272,14 @@ hash_table_find_next(const hash_table_t *h, ht_link_t *first, ht_link_t *item)
 	assert(h && h->bucket);
 
 	size_t idx = h->op->hash(item) % h->bucket_cnt;
+	list_t *list = &h->bucket[idx];
 
 	/* Traverse the circular list until we reach the starting item again. */
-	for (link_t *cur = item->link.next; cur != &first->link;
-	    cur = cur->next) {
-		assert(cur);
+	for (link_t *cur = list_next(&item->link, list); cur != &first->link;
+	    cur = list_next(cur, list)) {
 
-		if (cur == &h->bucket[idx].head)
-			continue;
+		if (!cur)
+			cur = list_first(list);
```

Comment:
This does not seem right. If 'first' is literally the first item on the list, the loop does not terminate on it (as it should) and the loop can create a livelock together with irq_dispatch_and_lock_table().